### PR TITLE
chore: fix typos

### PR DESCRIPTION
**Pull request by herluf-ba**
A humble contribution, but a contribution none the less!

Additionally I was confused about the following things as I read through the book:
- The `!>` operator is used without introduction. I think a natural place to mention it would be under "Functions" > "Pipelines".
- I needed to run nightly to run many of the code examples. Maybe, while Flix hasn't reached a stable version yet it would be good to mention nightly in one of the first sections. Something like this would have helped me:
> Flix is actively being developed and haven't reached a stable version yet. Meanwhile we want to keep this book as in line with cutting edge Flix, so it is recommended that you use the [nightly build](https://flix.dev/nightly/) to make sure that you are able to run all examples.

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Excellent! Thanks.

Can I ask you to add your name to `AUTHORS.md` to indicate you agree to make your contributions open source.

**Comment by magnus-madsen** [comment]
> I needed to run nightly to run many of the code examples. Maybe, while Flix hasn't reached a stable version yet it would be good to mention nightly in one of the first sections. 

We have just released a new stable release, so now all examples should work!

**Comment by magnus-madsen** [comment]
(Literally released this morning)

File: src/immutable-data.md [hunk 1]
```diff
@@ -20,7 +20,6 @@ Other immutable data types include:
 
 - `Option[t]`       : A type that is either `None` or `Some(t)`.
 - `Result[e, t]`    : A type that is either `Ok(t)` or `Err(e)`.
-- `Chain[t]`        : An immutable sequence of elements of type `t` that supports fast append.
```
Comment (magnus-madsen) [inline src/immutable-data.md:23]: Why delete this? :) Please put it back :)

Comment (herluf-ba) [reply src/immutable-data.md:23]: This is deleted since `Chain` is also mentioned in the list above, so it is listed twice

Comment (magnus-madsen) [reply src/immutable-data.md:23]: Oh. Well spotted. I must be blind :)

**Comment by magnus-madsen** [comment]
Thanks-- changes are automatically deployed every hour.

Please send more updates, as you find them :)

**Pull request merged**